zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java

```diff
@@ -436,6 +438,35 @@ public void testElectionFraud() throws IOException, InterruptedException {
         Assert.assertTrue("falseLeader never rejoins the quorum", foundFollowing);
     }
 
+    @Test
+    public void testBadMyId() throws Exception {
+        ClientBase.setupTestEnv();
+
+        final int CLIENT_PORT_QP1 = PortAssignment.unique();
+        List<Integer> invalidMyidList = new ArrayList<>();
+        invalidMyidList.add(-1);
+        invalidMyidList.add(0);
+        invalidMyidList.add(256);
+        invalidMyidList.add(1024);
+        for (Integer myidForCheck : invalidMyidList) {
+            String quorumCfgSection = "server.1=127.0.0.1:"
+                    + PortAssignment.unique() + ":" + PortAssignment.unique()
+                    + "\nserver." + myidForCheck + "=127.0.0.1:" + PortAssignment.unique() + ":"
+                    + PortAssignment.unique();
+
+            MainThread q1 = new MainThread(myidForCheck, CLIENT_PORT_QP1, quorumCfgSection);
```

Comment:
Do we really need to setup all of this stuff in order to test one simple validation?

Reply:
I think `MainThread` is not that big deal to use in tests, because it's more like an integration test rather than unit.